app/src/main/java/com/example/parkingsystemjava/mvp/model/ReservationModel.java

```diff
@@ -34,4 +41,24 @@ public void setStartDate(Calendar startDate) {
     public void setEndDate(Calendar endDate) {
         this.reservation.setEndDate(endDate);
     }
+
+    public boolean existsOverlap(Reservation reservation) {
+        if (reservation != null) {
+            List<Reservation> reservationList = database.getReservation(reservation.getParkingLots());
+            if (reservationList != null) {
+                for (Reservation reservationSearched : reservationList) {
+                    if (reservationSearched.getEndDate().after(reservation.getStartDate())) {
```

Comment:
are you sure that this validation is correct?

Reply:
Done. No, it was incomplete, I change the validation so that it checks with the start (list) / end (reservation) date as well